### PR TITLE
NIFI-13436 Retain queue for non-modified connections during MiNiFi fl…

### PR DESCRIPTION
**Pull request by ferencerdei**
…ow update

# Summary

[NIFI-13436](https://issues.apache.org/jira/browse/NIFI-13436)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java [hunk 1]
```diff
@@ -105,8 +112,9 @@ public boolean update(byte[] rawFlow) {
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("Attempting to update flow with content: \n{}", new String(rawFlow, UTF_8));
         }
-
+        Set<String> originalConnectionIds = emptySet();
         try {
+            originalConnectionIds = findAllExistingConnections(flowController.getFlowManager().getRootGroup()).stream().map(Connection::getIdentifier).collect(Collectors.toSet());
```
Comment (briansolo1985) [inline minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java:117]: I would add new lines after each stream call to make it more comprehensive:
```
            originalConnectionIds = findAllExistingConnections(flowController.getFlowManager().getRootGroup())
                .stream()
                .map(Connection::getIdentifier)
                .collect(Collectors.toSet());
```

File: minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java [hunk 2]
```diff
@@ -262,4 +275,18 @@ private <T> Optional<T> retry(Supplier<T> input, Predicate<T> predicate, int max
             }
         }
     }
+
+    private Set<String> findAllProposedConnectionIds(VersionedProcessGroup versionedProcessGroup) {
+        return versionedProcessGroup == null ? emptySet() : Stream.concat(
```
Comment (briansolo1985) [inline minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java:280]: I would add new lines after each stream call to make it more comprehensive:
```
        return versionedProcessGroup == null
            ? emptySet()
            : Stream.concat(
                versionedProcessGroup.getConnections().stream().map(VersionedConnection::getInstanceIdentifier),
                versionedProcessGroup.getProcessGroups().stream().map(this::findAllProposedConnectionIds).flatMap(Set::stream)
            )
            .collect(Collectors.toSet());
```

File: minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java [hunk 3]
```diff
@@ -262,4 +275,18 @@ private <T> Optional<T> retry(Supplier<T> input, Predicate<T> predicate, int max
...
+                versionedProcessGroup.getConnections().stream().map(VersionedConnection::getInstanceIdentifier),
+                versionedProcessGroup.getProcessGroups().stream().flatMap(childGroup -> findAllProposedConnectionIds(childGroup).stream())
+        ).collect(Collectors.toSet());
+    }
+
+    private Set<Connection> findAllExistingConnections(ProcessGroup processGroup) {
+        return processGroup == null ? emptySet() : Stream.concat(
```
Comment (briansolo1985) [inline minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java:287]: I would add new lines after each stream call to make it more comprehensive:
```
 return processGroup == null
            ? emptySet()
            : Stream.concat(
                processGroup.getConnections().stream(),
                processGroup.getProcessGroups().stream().map(this::findAllExistingConnections).flatMap(Set::stream)
            )
            .collect(Collectors.toSet());
    }
```

**Review by briansolo1985 — COMMENTED** [review]
Thanks for the improvement @ferencerdei 
The change looks good and it worked on my local setup
Left some minor cosmetic related comments, otherwise we should be good to go

**Comment by ferencerdei** [comment]
Thanks for the review, I've applied the requested changes

**Comment by briansolo1985** [comment]
Thanks for applying the recommended changes. LGTM +1. Will proceed with the merge

**Pull request closed without merging**